### ISSUE TITLE
Website does not load with 0 errors

### ISSUE THREAD

**Issue by XhiroYui** [HtmlUnit/htmlunit#551]
Running the below code causes the thread to get stuck (I've waited for several minutes) and nothing happens with no error message. 

```
private void website() {
        log.info("Line 1");
        try (final WebClient webClient = new WebClient(BrowserVersion.FIREFOX)) {
            try {
                final HtmlPage page1 = webClient.getPage("<REDACTED>");
                log.info("Line 2");
            } catch (IOException e) {
                throw new RuntimeException(e);
            }
        }
    }
 ```

With the above code, "Line 2" is never printed. Any idea on how to solve this or even debug this?

EDIT : after quite awhile I finally get Exception in thread "main" java.lang.OutOfMemoryError: Java heap space error
EDIT2 : Nvm, after disabled Javascript it works. Not sure which part of the JavaScript is causing this to completely hang
EDIT3 : Redacted the URL by request.

**Comment by RhestiaYuki** [comment]
Update : After turning off javscript, navigating down further pages is proving to be extremely troublesome. Is there a way to fix the JavaScript issue so I don't have to turn off JavaScript?

**Comment by mikomarrache** [comment]
Hi,

I have a similar issue. The call to `getPage` never returns. Same symptoms.

Any help would be appreciated. Completely stuck.

Thanks

**Comment by rbri** [comment]
see #552 for details

**Comment by rbri** [comment]
Made some progress here, hopefully this will be fixed with the next release.

**Comment by rbri** [comment]
Should be fixed in version 3.0.0 - please reopen if this still fails
